From 3ac4bfd9c19541c576d0804fde799eb4d50facdf Mon Sep 17 00:00:00 2001
From: Alexey Dobriyan <adobriyan@openvz.org>
Date: Fri, 7 Mar 2008 16:45:12 +0300
Subject: [PATCH 14/48] NETFILTER: move VE's expect_max value initialization into correct place

Copying from nf_ct_expect_max is pointless for VE0 because it's 0 at this place
Move initialization to nf_conntrack_expect_init() for correct behaviour.
---
 net/netfilter/nf_conntrack_core.c   |    1 -
 net/netfilter/nf_conntrack_expect.c |    2 +-
 2 files changed, 1 insertions(+), 2 deletions(-)

diff --git a/net/netfilter/nf_conntrack_core.c b/net/netfilter/nf_conntrack_core.c
index 46e0c5b..879dc9e 100644
--- a/net/netfilter/nf_conntrack_core.c
+++ b/net/netfilter/nf_conntrack_core.c
@@ -1167,7 +1167,6 @@ int nf_conntrack_init(void)
 	}
 
 	ve_nf_conntrack_max = nf_conntrack_max;
-	ve_nf_ct_expect_max = nf_ct_expect_max;
 	atomic_set(&ve_nf_conntrack_count, 0);
 	INIT_HLIST_HEAD(&ve_unconfirmed);
 #endif
diff --git a/net/netfilter/nf_conntrack_expect.c b/net/netfilter/nf_conntrack_expect.c
index c90024c..39dde85 100644
--- a/net/netfilter/nf_conntrack_expect.c
+++ b/net/netfilter/nf_conntrack_expect.c
@@ -521,7 +521,7 @@ int nf_conntrack_expect_init(void)
 			nf_ct_expect_hsize = 1;
 	}
 	nf_ct_expect_max = nf_ct_expect_hsize * 4;
-
+	ve_nf_ct_expect_max = nf_ct_expect_max;
 	ve_nf_ct_expect_hash = nf_ct_alloc_hashtable(&nf_ct_expect_hsize,
 						  &ve_nf_ct_expect_vmalloc);
 	if (ve_nf_ct_expect_hash == NULL)
-- 
1.5.4.3

